examples/csharp/App/Program.cs

```diff
@@ -138,29 +251,37 @@ private static void VerifyToken(String token, String responseBodyHash, SecurityK
             }
 
             var jwtToken = new JwtSecurityToken(token);
-            if (jwtToken.Subject != _subject)
+            if (jwtToken.Subject != Subject)
             {
                 throw new ApplicationException("invalid subject: " + jwtToken.Subject);
             }
         }
 
-        private static string CalculateRequestBodyHash(string body)
+        private static string CalculateRequestBodyHash(byte[] body)
         {
-            var bytes = Encoding.UTF8.GetBytes(body);
-            var hash = new SHA256Managed().ComputeHash(bytes);
-            return Convert.ToBase64String(hash);
+            if (body.Length == 0)
+            {
+                return string.Empty;
+            }

```

Comment:
It may be removed. I don’t know if it makes sense to leave this, because it will never be used in the current implementation